src/main/java/interaction/Interaction.java

```diff
@@ -12,11 +14,30 @@ public Interaction(GameObject first, GameObject second) {
         this.second = second;
     }
 
-    public GameObject getFirst() {
+    GameObject getFirst() {
         return first;
     }
 
-    public GameObject getSecond() {
+    GameObject getSecond() {
         return second;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Interaction that = (Interaction) o;
+        return (Objects.equals(first, that.first) &&
+                Objects.equals(second, that.second)) ||
+                (Objects.equals(first, that.second) &&
+                        Objects.equals(second, that.first));
+    }
+
+    @Override
+    public int hashCode() {
+        int res = 17;
+        res = res * 31 + Math.min(first.hashCode(), second.hashCode());
```

Comment:
GameObject hashCode is not overriden. This might lead to collision

Reply:
GameObject hashCode() has default implementation, isn't it enought?

Reply:
default implementation is not a good choice which might lead to hash collisions. Imagine that you use different JVMs and each of them its own hashCode() implementation.

Reply:
If you call hashCode() you should **always!** override it